src/wallet/api.rs

```diff
@@ -1399,11 +1359,11 @@ impl Wallet {
         pubkey1: &PublicKey,
         pubkey2: &PublicKey,
     ) -> Result<(), WalletError> {
-        self.import_multisig_redeemscript_descriptor(
-            pubkey1,
-            pubkey2,
-            &self.get_core_wallet_label(),
-        )
+        let descriptor = self
+            .rpc
+            .get_descriptor_info(&format!("wsh(sortedmulti(2,{},{}))", pubkey1, pubkey2))?
+            .descriptor;
+        self.import_descriptors(&[descriptor], None)
     }
```

Comment:
We should be able to remove these two functions now? We have a global `import_descriptors` and `collect_descriptors` API, and we don't care about reimporting anymore. So that should mean we do not have to care about importing specific multisig/contract redeemscripts. Everything will be reimported again in the next sync call.

This will allow us to remove these RPC dependent dangling functions from Maker and Taker business logic. This is a potential blocker we discussed for CBF syncing, and we now got the chance to fix it. 

Deleting this functions and adding `wallet.sync()` in the appropriate call sites should work. same for `import_tx_with_merkleproof`. We don't do that merkleproof importing anymore.